src/commands/schema/generate.ts

```diff
@@ -9,61 +9,84 @@ export type SchemasMap = {
 
   constructor(private base: SnapshotCommand, private ignorevoid = true) {}
 
-  public async generate(): Promise<SchemasMap> {
+  public async generate(): Promise<Schemas> {
     for (const cmd of this.base.commands) {
       // eslint-disable-next-line no-await-in-loop
       const loadedCmd = await cmd.load() // commands are loaded async in oclif/core
       this.classToId[loadedCmd.name] = loadedCmd.id
     }
 
-    const schemas: SchemasMap = {}
+    const cmdSchemas: SchemasMap = {}
 
     for (const file of this.getAllCmdFiles()) {
-      const {returnType, commandId} = this.parseFile(file)
+      const {returnType, commandId} = this.parseCmdFile(file)
       if (this.ignorevoid && returnType === 'void') continue
-      try {
-        const config = {
-          path: file,
-          type: returnType,
-          skipTypeCheck: true,
-        }
-        const schema = createGenerator(config).createSchema(config.type)
-        schemas[commandId] = schema
-      } catch (error: any) {
-        if (error.message.toLowerCase().includes('no root type')) {
-          throw new Error(`Schema generator could not find the ${red(returnType)} type. Please make sure that ${red(returnType)} is exported.`)
-        } else {
-          throw error
-        }
+      cmdSchemas[commandId] = this.generateSchema(returnType, file)
+    }
+
+    const hookSchemas: SchemasMap = {}
+    for (const file of this.getAllHookFiles()) {
+      const {returnType, hookId} = this.parseHookFile(file)
+      if (returnType && hookId) {
+        hookSchemas[hookId] = this.generateSchema(returnType, file)
       }
     }
 
-    return schemas
+    return {commands: cmdSchemas, hooks: hookSchemas}
   }
 
-  private getAllCmdFiles(dirPath: string = path.join(this.base.config.root, 'src', 'commands'), allFiles: string[] = []) {
-    const files = fs.readdirSync(dirPath)
-
-    files.forEach(file => {
-      const fPath = path.join(dirPath, file)
-      if (fs.statSync(fPath).isDirectory()) {
-        allFiles = this.getAllCmdFiles(fPath, allFiles)
-      } else if (file.endsWith('ts')) {
-        allFiles.push(fPath)
+  private generateSchema(returnType: string, file: string): Schema {
+    try {
+      const config = {
+        path: file,
+        type: returnType,
+        skipTypeCheck: true,
       }
-    })
+      return createGenerator(config).createSchema(config.type)
+    } catch (error: any) {
+      if (error.message.toLowerCase().includes('no root type')) {
+        throw new Error(`Schema generator could not find the ${red(returnType)} type. Please make sure that ${red(returnType)} is exported.`)
+      } else {
+        throw error
+      }
+    }
+  }
+
+  private getAllCmdFiles(): string[] {
+    return getAllFiles(path.join(this.base.config.root, 'src', 'commands'), '.ts')
+  }
 
-    return allFiles
+  private getAllHookFiles(): string[] {
+    return getAllFiles(path.join(this.base.config.root, 'src', 'hooks'), '.ts')
```

Comment:
Will the directory with oclif hooks always be named `hooks`?
From the [oclif docs](https://oclif.io/docs/hooks) it seems that can be user defined.